Makefile

```
@@ -90,6 +91,36 @@ build-client-dotnet:
 	make run-in-docker sdk_language=dotnet image=mcr.microsoft.com/dotnet/sdk:${DOTNET_DOCKER_TAG} command="/bin/sh -c 'dotnet format ./OpenFga.Sdk.sln'" || true
 	make run-in-docker sdk_language=dotnet image=mcr.microsoft.com/dotnet/sdk:${DOTNET_DOCKER_TAG} command="/bin/sh -c 'dotnet format ./OpenFga.Sdk.sln'"
 
+### Python
+.PHONY: tag-client-python
+tag-client-python: test-client-python
+	make utils-tag-client sdk_language=python
+
+.PHONY: test-client-python
+test-client-python: build-client-python
+	# Need to ignore F401 (unused module), E402 (import not at top of module), E501 (line too long) and W504 (line break after binary operator)
+	# due to limitations of autopep8 as well as generator. 
+	make run-in-docker sdk_language=python image=python:${PYTHON_DOCKER_TAG} command="/bin/sh -c 'python -m pip install -r test-requirements.txt; python -m unittest test/*; python -m flake8 --ignore F401,E402,E501,W504 openfga_sdk; python -m flake8 --ignore E501 test'"
+
+.PHONY: build-client-python
+build-client-python:
+	make build-client sdk_language=python tmpdir=${TMP_DIR}
+	# Update so that Python dictionary is defined correctly
+	sed -i -e "s|\"key\": |key=|g" ${CLIENTS_OUTPUT_DIR}/fga-python-sdk/README.md
+	sed -i -e "s|from openfga_sdk.model.tuple_keys import TupleKeys|from openfga_sdk.model.tuple_key import TupleKey\nfrom openfga_sdk.model.tuple_keys import TupleKeys|g" ${CLIENTS_OUTPUT_DIR}/fga-python-sdk/README.md
+	rm -rf  ${CLIENTS_OUTPUT_DIR}/fga-python-sdk/README.md-e
+	whoami
+	ls -la ${CLIENTS_OUTPUT_DIR}/fga-python-sdk
+	ls -la ${CLIENTS_OUTPUT_DIR}/fga-python-sdk/docs
+	sed -i -e "s|\"key\": |key=|g" ${CLIENTS_OUTPUT_DIR}/fga-python-sdk/docs/*.md
+	sed -i -e "s|from openfga_sdk.model.tuple_keys import TupleKeys|from openfga_sdk.model.tuple_key import TupleKey\nfrom openfga_sdk.model.tuple_keys import TupleKeys|g" ${CLIENTS_OUTPUT_DIR}/fga-python-sdk/docs/*.md
+	rm -rf ${CLIENTS_OUTPUT_DIR}/fga-python-sdk/docs/*.md-e
+	# The return value is falsely marked as not found
+	sed -i '' -e "s|-> 'relations':|-> 'relations': # noqa: F821|g" ${CLIENTS_OUTPUT_DIR}/fga-python-sdk/*/model/type_definition.py
+	rm -rf  ${CLIENTS_OUTPUT_DIR}/fga-python-sdk/*/model/type_definition.py-e
+	# Need to ignore E402 (import order) to avoid circular dependency
+	make run-in-docker sdk_language=python image=python:${PYTHON_DOCKER_TAG} command="/bin/sh -c 'python -m pip install autopep8; autopep8 --in-place --ignore E402 --recursive openfga_sdk; autopep8 --in-place --recursive test'"
+
 .PHONY: run-in-docker
 run-in-docker:
 	docker run --rm \
```

```suggestion
	docker run --rm \
    -u $(id -u ${USER}):$(id -g ${USER}) \
```

Could you try with this?

---

As discussed offline, we need to do it in build-client instead of run-in-docker because the build-client did not use run-in-docker.